File: apps/frontend/packages/api-services/src/forms/application/FormResponse.ts
import type { Response } from '@lyra/shared';
import type { Form } from '~/forms/domain/Form.ts';

export default class FormResponse implements Response {
	constructor(
		public readonly id: string,
		public readonly name: string,
		public readonly header: string,
		public readonly description: string,
		public readonly inputPlaceholder: string,
		public readonly buttonText: string,
		public readonly buttonColor: string,
		public readonly backgroundColor: string,
		public readonly textColor: string,
		public readonly buttonTextColor: string,
		public readonly createdAt: string | null = null,
		public readonly updatedAt: string | null = null
	) {}

	static from(form: Form): FormResponse {
		return new FormResponse(
			form.id,
			form.name,
			form.header,
			form.description,
			form.inputPlaceholder,
			form.buttonText,
			form.buttonColor,
			form.backgroundColor,
			form.textColor,
			form.buttonTextColor,
			form.createdAt,
			form.updatedAt
		);
	}
}

File: apps/backend/src/main/kotlin/com/lyra/app/forms/application/FormResponse.kt
package com.lyra.app.forms.application

import com.lyra.app.forms.domain.Form
import com.lyra.common.domain.bus.query.Response

/**
 * Data class representing a response for a form.
 *
 * @property id The unique identifier of the form.
 * @property name The name of the form.
 * @property header The header of the form.
 * @property description The description of the form.
 * @property inputPlaceholder The input placeholder of the form.
 * @property buttonText The text of the button.
 * @property buttonColor The color of the button.
 * @property backgroundColor The background color of the form.
 * @property textColor The text color of the form.
 * @property buttonTextColor The text color of the button.
 * @property organizationId The unique identifier of the organization.
 * @property createdAt The creation time of the form.
 * @property updatedAt The last update time of the form.
 */
data class FormResponse(
    val id: String,
    val name: String,
    val header: String,
    val description: String,
    val inputPlaceholder: String,
    val buttonText: String,
    val buttonColor: String,
    val backgroundColor: String,
    val textColor: String,
    val buttonTextColor: String,
    val organizationId: String,
    val createdAt: String? = null,
    val updatedAt: String? = null,
) : Response {
    companion object {
        /**
         * Creates a new FormResponse.kt from a given Form.
         *
         * @param form The form to create the response from.
         * @return The newly created FormResponse.kt.
         */
        fun from(form: Form) = FormResponse(
            id = form.id.value.toString(),
            name = form.name,
            header = form.header,
            description = form.description,
            inputPlaceholder = form.inputPlaceholder,
            buttonText = form.buttonText,
            buttonColor = form.buttonColor.toString(),
            backgroundColor = form.backgroundColor.toString(),
            textColor = form.textColor.toString(),
            buttonTextColor = form.buttonTextColor.toString(),
            organizationId = form.organizationId.value.toString(),
            createdAt = form.createdAt.toString(),
            updatedAt = form.updatedAt?.toString(),
        )
    }
}
